service/biz/enrollz_biz.go

```diff
@@ -135,3 +159,76 @@ func EnrollControlCard(controlCardSelection *cpb.ControlCardSelection, deps Enro
 	// Return a successful (no-error) response.
 	return nil
 }
+
+// Request to RotateOwnerIakCert().
+type RotateOwnerIakCertReq struct {
+	// Selection of a specific switch control card.
+	controlCardSelection *cpb.ControlCardSelection
+	// Infra-specific wired dependencies.
+	deps EnrollzInfraDeps
+	// Verification options for IAK cert.
+	certVerificationOpts x509.VerifyOptions
+}
+
+// This is a "client"/switch-owner side implementation of Enrollz service. This client-side logic
+// is part of the switch owner infra/service and is expected to communicate with a device/switch (hosting
+// enrollz gRPC endpoints) to verify its TPM-based identity and provision it with the switch-owner-issued,
+// TPM-based attestation Owner IAK cert. Switch owner is expected to TPM-enroll one switch control card
+// at a time, starting with an active card.
+//
+// More specifically, this function targets rotation of the Owner IAK cert in a post-install scenario.
+// This means that the switch may NOT be using (o/)IDevID TLS certs anymore, and instead relies on a "real"
+// prod mTLS cert that was issued to the switch after its first successful attestation. Thus, this function
+// only assumes the presence of an IAK cert in the response from the device. Consumers of this function
+// should carefully choose CA trust anchors in `x509.VerifyOptions` to include switch vendor CA root and
+// intermediate certs that signed the IAK cert. By the end of the workflow a given control card will obtain a
+// new (rotated) Owner IAK cert (signed by the switch owner CA).
+func RotateOwnerIakCert(req *RotateOwnerIakCertReq) error {
+	// 1. Call device's GetIakCert API for the specified control card.
+	getIakCertReq := &epb.GetIakCertRequest{ControlCardSelection: req.controlCardSelection}
+	getIakCertResp, err := req.deps.GetIakCert(getIakCertReq)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve IAK cert from the device with req=%s: %w",
```

Comment:
nit: capitalize first word: s/failed/Failed

Reply:
Actually, out internal style guide recommends to do otherwise: go/go-style/decisions#error-strings :)